src/Servers/Kestrel/Core/src/Internal/KestrelServerImpl.cs

```diff
@@ -276,14 +276,20 @@ private async Task BindAsync(CancellationToken cancellationToken)
                 Options.ConfigurationLoader?.Load();
 
                 await AddressBinder.BindAsync(Options.ListenOptions, AddressBindContext!).ConfigureAwait(false);
-                _configChangedRegistration = reloadToken?.RegisterChangeCallback(async state => await ((KestrelServerImpl)state).RebindAsync(), this);
+                _configChangedRegistration = reloadToken?.RegisterChangeCallback(Rebind, this);
             }
             finally
             {
                 _bindSemaphore.Release();
             }
         }
 
+        private static void Rebind(object state)
+        {
+            var server = (KestrelServerImpl)state;
+            _ = server.RebindAsync();
```

Comment:
Nit: Maybe rename this to `TriggerRebind`.

I also think we should add a comment noting that the _bindSemaphore prevents RebindAsync from interfering with shutdown or other rebinds and that all exceptions are caught and logged at the critical level.

This deserved a comment before, but better late than never. I do like that this makes it more obvious that the RebindAsync Task isn't being awaited.